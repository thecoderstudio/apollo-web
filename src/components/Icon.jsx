import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const propTypes = {
  active: PropTypes.bool
}

const defaultProps = {
  active: true
}

const Icon = styled.i`
  cursor: ${props => props.active ? 'pointer' : 'inherit'};
`;

Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;

export default Icon
