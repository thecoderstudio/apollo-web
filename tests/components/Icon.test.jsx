import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../../src/components/Icon';

describe('icon', () => {
  it('renders active correctly', () => {
    const tree = renderer.create(<Icon active />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders inactive correctly', () => {
    const tree = renderer.create(<Icon active={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
