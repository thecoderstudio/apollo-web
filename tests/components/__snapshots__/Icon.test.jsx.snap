// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon renders active correctly 1`] = `
.c0 {
  cursor: pointer;
}

<i
  className="c0"
/>
`;

exports[`icon renders inactive correctly 1`] = `
.c0 {
  cursor: inherit;
}

<i
  className="c0"
/>
`;
